pkg/advisory/create.go

```diff
@@ -91,3 +92,38 @@ func createAdvisoryConfig(ctx context.Context, documents *configs.Index[v2.Docum
 
 	return nil
 }
+
+func GenerateCGAID(packageName, vulnerabilityID string) (string, error) {
+	// Concatenate package name and VulnerabilityID
+	inputString := packageName + vulnerabilityID
+
+	// Generate SHA-256 hash of the input string
+	hash := sha256.Sum256([]byte(inputString))
+	hashHex := hex.EncodeToString(hash[:])
+
+	// Allowed characters [23456789cfghjmpqrvwx]
+	allowedChars := "23456789cfghjmpqrvwx"
+
+	// Create a map from hex characters to allowed characters
+	hexChars := "0123456789abcdef"
+	charMap := make(map[byte]byte)
+	for i := 0; i < len(hexChars); i++ {
+		charMap[hexChars[i]] = allowedChars[i%len(allowedChars)]
+	}
+
+	// Replace each hex digit with corresponding allowed character
+	var customUUID []byte
+	for i := 0; i < len(hashHex); i++ {
+		customUUID = append(customUUID, charMap[hashHex[i]])
+	}
+
+	// Ensure the string is long enough
+	if len(customUUID) < 12 {
+		return "", fmt.Errorf("hash length is insufficient to generate the UUID with the desired pattern")
+	}
+
+	// Format the custom UUID to match CGA(-[23456789cfghjmpqrvwx]{4}){3}
+	formattedUUID := fmt.Sprintf("CGA-%s-%s-%s", customUUID[0:4], customUUID[4:8], customUUID[8:12])
+
+	return formattedUUID, nil
+}
```

Comment:
Small nit: I'd move this code to a dedicated file so that it's easier to find!

Reply:
done